src/repo.rs

```diff
@@ -34,12 +34,50 @@ unsafe fn from_glib_container_variant_set(ptr: *mut glib_sys::GHashTable) -> Has
     set
 }
 
+/// An open transaction in the repository.
+///
+/// This will automatically invoke [`ostree::Repo::abort_transaction`] when the value is dropped.
+pub struct TransactionGuard<'a> {
+    /// Reference to the repository for this transaction.
+    repo: Option<&'a Repo>,
```

Comment:
I think the C counterpart follows a different logic, and always call abort (even on successful commit).

One additional confusion I have is related to lifetime and ownership. I'm personally fine with the reference here, but I think everywhere we are following the mantra that `Repo` is refcounted internally.

Reply:
https://github.com/ostreedev/ostree/issues/2413#issuecomment-930327836

Reply:
Right, it's safe to call abort, but I optimized this a bit to not do so if it's not necessary, because I think the logic is cleaner.

> I'm personally fine with the reference here, but I think everywhere we are following the mantra that Repo is refcounted internally.

True, but I think the "eh let's just bump the refcount" thing comes into play when we're doing things like passing a repo instance across worker threads (i.e. `tokio::task::spawn_blocking`).  

In the PR where I removed the lifetime in some of the tar code, it was partly because of all the extra noise on the repo lifetime across many methods.

Here there's only a tiny struct with basically one method, we easily bind the lifetime of the guard to the repo, I think it's better to do so.

Reply:
Yes, :+1:  from my side.
As noted in the other comment I also like this more, and should probably tweak the C logic in this direction.